### PR TITLE
Update UAP System.Private.CoreLib to unblock UAP builds

### PR DESCRIPTION
**Pull request by nattress**
System.Private.CoreLib is currently in a torn state in CoreCLR vs CoreRT where the signature for `Stream.WriteAsync` has changed. This PR consumes a hand-built targeting pack containing System.Private.CoreLib's pending changes we want to merge into TFS. This should allow us to unblock the uap build and get packages we can consume in TFS. Once that happens, we can merge CoreRT into TFS, and then update with a fixed targeting pack.

### CODE REVIEW

**Comment by nattress** [comment]
@MichalStrehovsky @jkotas FYI

File: src/System.Threading.Tasks.Extensions/pkg/System.Threading.Tasks.Extensions.pkgproj [hunk 1]
```diff
@@ -12,6 +12,7 @@
     <ProjectReference Include="..\src\System.Threading.Tasks.Extensions.csproj" />
 
     <InboxOnTargetFramework Include="netcoreapp2.1" />
+    <InboxOnTargetFramework Include="$(UAPvNextTFM)" />
```
Comment (nattress) [inline src/System.Threading.Tasks.Extensions/pkg/System.Threading.Tasks.Extensions.pkgproj]: @stephentoub I had to revert this part of your recent commit https://github.com/dotnet/corefx/commit/c37e3e5f6b2a3fce69cffe1096ff5c491ed3639d as it broke the uap build.

Comment (stephentoub) [reply src/System.Threading.Tasks.Extensions/pkg/System.Threading.Tasks.Extensions.pkgproj]: @weshaggard or @ericstj ?

**Comment by nattress** [comment]
@dotnet-bot Test UWP CoreCLR x64 Debug Build

**Comment by danmosemsft** [comment]
The Linux failure is preexisting - XmlSchemaValidatorApi.Tests

File: src/System.Threading.Tasks.Extensions/pkg/System.Threading.Tasks.Extensions.pkgproj (hunk 1 above)
Comment (ericstj) [reply src/System.Threading.Tasks.Extensions/pkg/System.Threading.Tasks.Extensions.pkgproj]: This isn't a revert.  @nattress [updated the package index](https://github.com/dotnet/corefx/pull/27805/files#diff-122916076db7087dbc454352fada61eeR4904) to make this inbox (correctly) on UAP**vNext** (AKA uap10.0.16300) just like its inbox on NETCoreApp**vNext** (netcoreapp2.1).
Note that @joperezr will also be updating the package index to correctly account for the old assembly version shipping in the previous version of UAP.  https://github.com/dotnet/corefx/pull/27787

File: external/test-runtime/XUnit.Runtime.depproj [hunk 2]
```diff
@@ -182,10 +182,16 @@
           SkipUnchangedFiles="true"/>
   </Target>
 
+<!--
+  
+  Disable this check until we get the CoreCLR and TFS versions of System.Private.CoreLib back in sync after recent breaking changes.
```
Comment (joperezr) [inline external/test-runtime/XUnit.Runtime.depproj]: I'm not sure I understand why do we want to comment this out. We should instead just update both of the PN packages to the same version. Is there not a new TestILC from the same build available?

File: pkg/Microsoft.Private.PackageBaseline/packageIndex.json [hunk 3]
```diff
@@ -4901,7 +4901,7 @@
       "InboxOn": {
         "netcoreapp2.0": "4.1.1.0",
         "netcoreapp2.1": "4.2.0.0",
-        "uap10.0.16300": "4.1.2.0"
+        "uap10.0.16300": "4.2.0.0"
```
Comment (joperezr) [inline pkg/Microsoft.Private.PackageBaseline/packageIndex.json]: It's weird this isn't flagged as a conflict

File: src/System.Threading.Tasks.Extensions/pkg/System.Threading.Tasks.Extensions.pkgproj (hunk 1 above)
Comment (joperezr) [reply src/System.Threading.Tasks.Extensions/pkg/System.Threading.Tasks.Extensions.pkgproj]: those changes are in on master now

File: external/test-runtime/XUnit.Runtime.depproj (hunk 2 above)
Comment (nattress) [reply external/test-runtime/XUnit.Runtime.depproj]: We turned the targeting pack off a few days ago but not TestIlc - not realizing this check exists. So they won't be back in sync until we get a UAP build out of CoreFX.

File: pkg/Microsoft.Private.PackageBaseline/packageIndex.json (hunk 3 above)
Comment (nattress) [reply pkg/Microsoft.Private.PackageBaseline/packageIndex.json]: Unsure what you mean here. Do each of these entries need unique version numbers?

Comment (joperezr) [reply pkg/Microsoft.Private.PackageBaseline/packageIndex.json]: No, I just meant that I added some changes on master on these lines as well so I would have expected that this change would have caused a conflict. The change itself is fine.

**Pull request merged**